### ISSUE TITLE
Error handling routing seems broken

### ISSUE THREAD

**Issue by PVince81** [owncloud/guests#130]
Copied from @davitol's report to the password_policy app:

Using https://github.com/owncloud/guests/pull/109

**Steps to reproduce**
1. Set a policy of minimum 8 characters.
2. Add a new guest user via email
3. Set a password of less than 8 characters for the guest user.

**Expected behavior**

A message about wrong password for the policy should be shown

**Current behavior**

![screen shot 2017-07-18 at 13 17 38](https://user-images.githubusercontent.com/10596560/28314730-fd7b6912-6bbb-11e7-94fb-fd56895c00f4.png)

![screen shot 2017-07-18 at 13 17 31](https://user-images.githubusercontent.com/10596560/28314736-0437414a-6bbc-11e7-8995-3b942a49f38f.png)

owncloud log:

```

{"reqId":"Bo7j1leRVZla2BpKswiX","level":3,"time":"2017-07-18T11:17:25+00:00","remoteAddr":"90.173.169.6","user":"admin","app":"PHP","method":"GET","url":"\/index.php\/apps\/guests\/register\/dtoledo@solidgear.es\/usQpumHDZdqBMmThk0lDT","message":"Undefined index: user_autofocus at \/opt\/owncloud\/apps\/guests\/templates\/form.password.php#22"}
{"reqId":"pxpOHldiN1PLCiYQKwfH","level":3,"time":"2017-07-18T11:17:28+00:00","remoteAddr":"90.173.169.6","user":"admin","app":"index","method":"POST","url":"\/index.php\/apps\/guests\/register","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Call to undefined method OCA\\\\Guests\\\\Controller\\\\RegisterController::error()\",\"Code\":0,\"Trace\":\"#0 [internal function]: OCA\\\\Guests\\\\Controller\\\\RegisterController->register()\\n#1 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(159): call_user_func_array(Array, Array)\\n#2 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Guests\\\\Controller\\\\RegisterController), 'register')\\n#3 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(98): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Guests\\\\Controller\\\\RegisterController), 'register')\\n#4 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\Guests\\\\\\\\Cont...', 'register', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#5 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#6 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#7 \\\/opt\\\/owncloud\\\/lib\\\/base.php(920): OC\\\\Route\\\\Router->match('\\\/apps\\\/guests\\\/re...')\\n#8 \\\/opt\\\/owncloud\\\/index.php(49): OC::handleRequest()\\n#9 {main}\",\"File\":\"\\\/opt\\\/owncloud\\\/apps\\\/guests\\\/controller\\\/registercontroller.php\",\"Line\":226}"}
```

**Comment by PVince81** [comment]
expanding the exception:
```
{"reqId":"pxpOHldiN1PLCiYQKwfH","level":3,"time":"2017-07-18T11:17:28+00:00","remoteAddr":"90.173.169.6","user":"admin","app":"index","method":"POST","url":"\/index.php\/apps\/guests\/register","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Call to undefined method OCA\\\\Guests\\\\Controller\\\\RegisterController::error()\",\"Code\":0,\"Trace\":\"
#0 [internal function]: OCA\\\\Guests\\\\Controller\\\\RegisterController->register()\\n
#1 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(159): call_user_func_array(Array, Array)\\n
#2 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Guests\\\\Controller\\\\RegisterController), 'register')\\n
#3 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(98): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Guests\\\\Controller\\\\RegisterController), 'register')\\n
#4 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\Guests\\\\\\\\Cont...', 'register', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n
#5 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n
#6 \\\/opt\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n
#7 \\\/opt\\\/owncloud\\\/lib\\\/base.php(920): OC\\\\Route\\\\Router->match('\\\/apps\\\/guests\\\/re...')\\n
#8 \\\/opt\\\/owncloud\\\/index.php(49): OC::handleRequest()\\n#9 {main}\",\"File\":\"\\\/opt\\\/owncloud\\\/apps\\\/guests\\\/controller\\\/registercontroller.php\",\"Line\":226}"}

```

The error is in the guests app, apparently the error handling wasn't implemented correctly.

**Comment by PVince81** [comment]
@cortho is this something you could look at ?

**Comment by cortho** [comment]
@PVince81 oh shame. I found the mistake, however I don't have access to the password policy app.
Does it throw an exception, so I can get the error message with $e->getMessage()?

**Comment by PVince81** [comment]
It is very likely that the app is throwing an exception, yes. You should be able to get the message from there.

**Comment by cortho** [comment]
@PVince81 changed the broken code, hope this is fixed now

**Comment by cortho** [comment]
fixed in #109

**Comment by davitol** [comment]
Tested and it is solved 👍
